src/tools/run-make-support/src/diff/mod.rs

```diff
@@ -0,0 +1,64 @@
+use similar::TextDiff;
+use std::path::Path;
+
+#[cfg(test)]
+mod tests;
+
+pub fn diff() -> Diff {
+    Diff::new()
+}
+
+#[derive(Debug)]
+pub struct Diff {
+    expected: Option<String>,
+    actual: Option<String>,
+}
```

Comment:
I think it's easier for debugging if we are explicit about the diff headers, as in if it comes from a file then the expected header name can just be the file, if it comes from somewhere else, then ask the user to provide an expected header name like `expected_text("custom_name", "foo")`.